### PR TITLE
ddl: data type default values support `STR_TO_DATE('1980-01-01','%Y-%m-%d')`

### PR DESCRIPTION
**Pull request by zimulala**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/51276

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/ddl/ddl_api.go [hunk 1]
```diff
@@ -1365,6 +1365,17 @@ func getFuncCallDefaultValue(col *table.Column, option *ast.ColumnOption, expr *
 			}
 		}
 		return nil, false, dbterror.ErrDefValGeneratedNamedFunctionIsNotAllowed.GenWithStackByArgs(col.Name.String(), expr.FnName.String())
+	case ast.StrToDate:
+		// Support STR_TO_DATE('1980-01-01', '%Y-%m-%d').
+		if err := expression.VerifyArgsWrapper(expr.FnName.L, len(expr.Args)); err != nil {
```
Comment (Defined2014) [inline pkg/ddl/ddl_api.go:1370]: How about args not a const? Should we support it?

Comment (zimulala) [reply pkg/ddl/ddl_api.go:1370]: Updated.

**Review by Defined2014 — APPROVED** [review]

**Comment by zimulala** [comment]
UT  failed because encountered this issue: https://github.com/pingcap/tidb/issues/51289.

**Review by CbcWestwolf — APPROVED** [review]

**Review by tangenta — APPROVED** [review]

**Pull request merged**